docker-compose.yml

```diff
@@ -0,0 +1,30 @@
+version: "3"
+services:
+  node-dev:
+    user: node_user
+    build: 
+      context: .
+      dockerfile: Node.Dockerfile
+    volumes:
```

Comment:
Estou sentindo falta de um volume declarado exclusivamente para o `node_modules`

Reply:
O volume da aplicação está inteiro mapeado para dentro do container. Acho que não precisamos mapear tudo explicitamente.

Reply:
Mas isso irá bindar o `node_modules` da máquina com a do container. E se o usuário tiver um `node_modules` como vai ficar?

Reply:
Acho que gerou uma confusão. O docker é parte do ambiente de desenvolvimento, servindo apenas para ter o ambiente do node em um lugar separado e já disponível. Se separarmos os ambientes, a cada mudança vai demandar um rebuild no docker. Não foi essa a intenção do container.

Reply:
Entendo a colocação. 

O código em si está no volume `-.:/app` e não precisa de um build para atualizar.

No meu caso, uso python no container docker e realmente dou build novamente para instalar as dependências do projeto. Pois gosto de deixar separado as libs da minha máquina com as libs do container.

Se não for desejável esse isolamento, podemos resolver essa discussion.

Reply:
Entendo a colocação... os módulos globais precisam estar no docker (por exemplo, o nodemon fica lá), mas os locais são compartilhados. Por exemplo, estou criando uma nova classe, preciso de uma biblioteca nova (sei lá, o body-parser, por exemplo). Se estiver separado, iria precisar recriar o container.

Reply:
Ter o docker como ambiente de desenvolvimento pode ser problemático principalmente por questão de debug, algumas IDE's tem bastante delay quando vão usar o docker para fazer isso. Imagino que o docker vai nos ajudar nesse caso para subir as dependencias (banco, fila, etc)  de um jeito mais simples, para o ambiente de dev e o node que estara dentro do docker será apenas para rodar alguns comandos específicos (subir aplicação, testar), caso o dev não queira ter o node instalado na sua máquina.

Dessa forma, acho que vale ter o rebuild a cada vez que for subir a API no docker, para reinstalar as dependencias do node.